core-api/src/main/java/com/optimizely/ab/OptimizelyUserContext.java

```diff
@@ -0,0 +1,186 @@
+     * </ul>
+     * @param key A flag key for which a decision will be made.
+     * @param options A list of options for decision-making.
+     * @return A decision result.
+     */
+    public OptimizelyDecision decide(@Nonnull String key,
+                                     @Nonnull List<OptimizelyDecideOption> options) {
+        return optimizely.decide(this, key, options);
+    }
+
+    /**
+     * Returns a decision result ({@link OptimizelyDecision}) for a given flag key and a user context, which contains all data required to deliver the flag.
+     *
+     * @param key A flag key for which a decision will be made.
+     * @return A decision result.
+     */
+    public OptimizelyDecision decide(String key) {
+        return optimizely.decide(this, key, Collections.emptyList());
+    }
+
+    /**
+     * Returns a key-map of decision results ({@link OptimizelyDecision}) for multiple flag keys and a user context.
+     * <ul>
+     * <li>If the SDK finds an error for a key, the response will include a decision for the key showing <b>reasons</b> for the error.
+     * <li>The SDK will always return key-mapped decisions. When it can not process requests, it’ll return an empty map after logging the errors.
+     * </ul>
+     * @param keys A list of flag keys for which decisions will be made.
+     * @param options A list of options for decision-making.
+     * @return All decision results mapped by flag keys.
+     */
+    public Map<String, OptimizelyDecision> decideForKeys(@Nonnull List<String> keys,
+                                                         @Nonnull List<OptimizelyDecideOption> options) {
+        return optimizely.decideForKeys(this, keys, options);
+    }
+
+    /**
+     * Returns a key-map of decision results for multiple flag keys and a user context.
+     *
+     * @param keys A list of flag keys for which decisions will be made.
+     * @return All decision results mapped by flag keys.
+     */
+    public Map<String, OptimizelyDecision> decideForKeys(@Nonnull List<String> keys) {
+        return optimizely.decideForKeys(this, keys, Collections.emptyList());
+    }
+
+    /**
+     * Returns a key-map of decision results ({@link OptimizelyDecision}) for all active flag keys.
+     *
+     * @param options A list of options for decision-making.
+     * @return All decision results mapped by flag keys.
+     */
+    public Map<String, OptimizelyDecision> decideAll(@Nonnull List<OptimizelyDecideOption> options) {
+        return optimizely.decideAll(this, options);
+    }
+
+    /**
+     * Returns a key-map of decision results ({@link OptimizelyDecision}) for all active flag keys.
+     *
+     * @return A dictionary of all decision results, mapped by flag keys.
+     */
+    public Map<String, OptimizelyDecision> decideAll() {
+        return optimizely.decideAll(this, Collections.emptyList());
```

Comment:
same as above.

Reply:
Will keep it to save one-step indirection.